src/BlazorWebView/src/core/MauiAssetFileProvider.Windows.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Primitives;
+using Windows.ApplicationModel;
+using Windows.Storage;
+using FileAttributes = System.IO.FileAttributes;
+
+namespace Microsoft.AspNetCore.Components.WebView.Maui
+{
+	public partial class MauiAssetFileProvider
+	{
+		IDirectoryContents PlatformGetDirectoryContents(string subpath)
+			=> new WindowsMauiAssetDirectoryContents(subpath);
+
+		IFileInfo? PlatformGetFileInfo(string subpath)
+		{
+			// TODO: HACK: For now we strip out the folder because it is not preserved in the WinUI assets. We need to address this in the MauiAsset targets.
+			var fileWithoutFolders = Path.GetFileName(subpath);
+
+			// TODO: Also, instead of using WinUI Package APIs, we're going directly to the filesystem. They all seem to throw. Maybe we're on the wrong thread?
```

Comment:
@Redth - also FYI on this. Using the package APIs, which are all async, e.g. `GetItemAsync()`, would all throw mysterious C++ exceptions. I wonder if I'm running it on the wrong thread? (Which would be annoying, because nothing else here is async, and calling `GetResult()` is awful.)

Did you ever see any exceptions while using those APIs?